src/main/scala/se/nimsa/sbx/app/JsonFormats.scala

```diff
@@ -42,62 +42,91 @@ import se.nimsa.sbx.seriestype.SeriesTypeProtocol._
 import se.nimsa.sbx.storage.StorageProtocol._
 import se.nimsa.sbx.user.UserProtocol._
 
+import scala.util.{Failure, Success}
+
 trait JsonFormats {
 
   private def enumFormat[A](f: String => A) = Format(Reads[A] {
     case JsString(string) => JsSuccess(f(string))
     case _ => JsError("Enumeration expected")
   }, Writes[A](a => JsString(a.toString)))
 
-  private lazy val tagPathTrunkReads: Reads[TagPathTrunk] = (
-    (__ \ "tag").read[Int] and
-      (__ \ "item").read[String] and
-      (__ \ "previous").lazyReadNullable[TagPathTrunk](tagPathTrunkReads)
-    ) ((tag, itemString, previous) => {
-    val item = try Option(Integer.parseInt(itemString)) catch {
-      case _: Throwable => None
+  implicit lazy val tagPathReads: Reads[TagPath] =
```

Comment:
One major difference here is that the json representation now includes a `name` field next to `tag` which includes the name of the tag if it exists in the dictionary. When reading json either `tag` or `name` or both can be supplied. This makes handling of tag paths a lot easier on the client side and avoids constant server calls to get the keyword for a tag.